[BUG] org.opensearch.index.query.TermsQueryBuilderTests.testToQuery is flaky

### Describe the bug

```
java.lang.AssertionError: 
Expected: ((((((an instance of org.apache.lucene.search.TermInSetQuery or an instance of org.apache.lucene.search.PointInSetQuery) or an instance of org.apache.lucene.search.ConstantScoreQuery) or an instance of org.apache.lucene.search.MatchNoDocsQuery) or an instance of org.apache.lucene.search.IndexOrDocValuesQuery) or an instance of org.apache.lucene.search.MatchAllDocsQuery) or an instance of org.apache.lucene.search.FieldExistsQuery)
     but: was <mapped_boolean:F>
```

### Related component

Build

### To Reproduce

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.query.TermsQueryBuilderTests.testToQuery" -Dtests.seed=B65CA32A2CA0A9AC -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=da -Dtests.timezone=Asia/Urumqi -Druntime.java=21
NOTE: leaving temporary files on disk at: /var/jenkins/workspace/gradle-check/search/server/build/testrun/test/temp/org.opensearch.index.query.TermsQueryBuilderTests_B65CA32A2CA0A9AC-001
NOTE: test params are: codec=Asserting(Lucene99): {}, docValues:{}, maxPointsInLeafNode=901, maxMBSortInHeap=6.102524463741149, sim=Asserting(RandomSimilarity(queryNorm=true): {}), locale=da, timezone=Asia/Urumqi
NOTE: Linux 5.15.0-1056-aws amd64/Eclipse Adoptium 21.0.3 (64-bit)/cpus=32,threads=1,free=386066424,total=536870912
NOTE: All tests run in this JVM: [SpecialPermissionTests, AddVotingConfigExclusionsRequestTests, NodeInfoTests, VerifyRepositoryResponseTests, SnapshotIndexShardStatusTests, GetStoredScriptResponseTests, TransportVerifyShardBeforeCloseActionTests, GetIndexActionTests, RolloverRequestTests, DeleteComposableIndexTemplateRequestTests, ListViewNamesResponseTests, TransportShardBulkActionTests, IndexRequestBuilderTests, TransportResyncReplicationActionTests, QueryPhaseResultConsumerTests, SearchScrollRequestTests, RefreshPolicyTests, TransportWriteActionForIndexingPressureTests, OpenSearchUncaughtExceptionHandlerTests, FeatureAwareTests, ClusterBootstrapServiceRenamedSettingTests, NoClusterManagerBlockServiceRenamedSettingTests, AliasMetadataTests, HumanReadableIndexSettingsTests, MetadataMappingServiceTests, GroupShardsIteratorTests, AllocateUnassignedDecisionTests, DiskThresholdSettingsTests, PreferPrimaryAllocationTests, SingleShardNoReplicasRoutingTests, TargetPoolAllocationDeciderTests, FieldMemoryStatsTests, RemoteTransferContainerTests, DefaultCacheStatsHolderTests, GeoJsonShapeParserTests, PointBuilderTests, LoggingOutputStreamTests, NetworkModuleTests, DateFormattersTests, CollectionUtilsTests, AbstractLifecycleRunnableTests, RunOnceTests, JsonVsSmileTests, BlockingClusterStatePublishResponseHandlerTests, DiscoveryExtensionNodeTests, RestSendToExtensionActionTests, ReplicaShardAllocatorTests, IndexingPressureTests, AnalysisRegistryTests, FuzzyFilterPostingsFormatTests, SegmentTests, ReplaceMissingTests, CopyToMapperTests, DynamicTemplateTests, GeoShapeFieldTypeTests, MapperTests, PathMatchDynamicTemplateTests, BoolQueryBuilderTests, GeoBoundingBoxQueryBuilderTests, MoreLikeThisQueryBuilderTests, ScoreModeTests, TermsQueryBuilderTests]
```

### Expected behavior

Tests pass

### Additional Details

**Plugins**
Please list all plugins currently enabled.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - OS: [e.g. iOS]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/server/src/test/java/org/opensearch/index/query/TermsQueryBuilderTests.java
+++ b/server/src/test/java/org/opensearch/index/query/TermsQueryBuilderTests.java
@@ -41,6 +41,7 @@ import org.apache.lucene.search.MatchNoDocsQuery;
 import org.apache.lucene.search.PointInSetQuery;
 import org.apache.lucene.search.Query;
 import org.apache.lucene.search.TermInSetQuery;
+import org.apache.lucene.search.TermQuery;
 import org.apache.lucene.util.BytesRef;
 import org.opensearch.OpenSearchException;
 import org.opensearch.action.get.GetRequest;
@@ -141,6 +142,7 @@ public class TermsQueryBuilderTests extends AbstractQueryTestCase<TermsQueryBuil
                     .or(instanceOf(IndexOrDocValuesQuery.class))
                     .or(instanceOf(MatchAllDocsQuery.class))
                     .or(instanceOf(FieldExistsQuery.class))
+                    .or(instanceOf(TermQuery.class))
             );
             if (query instanceof ConstantScoreQuery) {
                 assertThat(((ConstantScoreQuery) query).getQuery(), instanceOf(BooleanQuery.class));

